lib/semantics/resolve-names.cc

```diff
@@ -1252,6 +1255,36 @@ std::optional<const DeclTypeSpec> ScopeHandler::GetImplicitType(
   return type;
 }
 
+// Convert symbol to be a ObjectEntity or return false if it can't be.
+bool ScopeHandler::ConvertToObjectEntity(Symbol &symbol) {
+  if (symbol.has<ObjectEntityDetails>()) {
+    // nothing to do
+  } else if (symbol.has<UnknownDetails>()) {
+    symbol.set_details(ObjectEntityDetails{});
+  } else if (auto *details{symbol.detailsIf<EntityDetails>()}) {
```

Comment:
Oh, this is the modern version of
if(a = b) {
...
}
What is the scope of details?